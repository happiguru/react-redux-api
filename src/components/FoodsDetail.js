import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { updateFoods } from '../actions/Foods';
import '../styles/FoodDetail.css';

const FoodsDetail = ({ food }) => (
  <div className="container mx-auto">
    <div className="row my-2">
      <div className="col-12 col-md-6">
        <img className="img-responsive mx-auto single-image" src={food?.strMealThumb} alt={food?.strMeal} />
      </div>
      <div className="col-12 col-md-6">
        <div className="recepies_info">
          <h3>{food?.strMeal}</h3>
          <p>{food?.strArea}</p>
          <p>{food?.strCategory}</p>
          <div className="resepies_details">
            <h4>Ingredients:</h4>
            <ul>
              <li>
                <p>
                  <strong>Ingredient 1 : </strong>
                  {food?.strIngredient1}
                </p>
              </li>
              <li>
                <p>
                  <strong>Ingredient 2 : </strong>
                  {food?.strIngredient2}
                </p>
              </li>
              <li>
                <p>
                  <strong>Ingredient 3 : </strong>
                  {food?.strIngredient3}
                </p>
              </li>
              <li>
                <p>
                  <strong>Ingredient 4 : </strong>
                  {food?.strIngredient4}
                </p>
              </li>
            </ul>
          </div>
          <a className="btn btn-info" href={food?.strYoutube} target="_blank" rel="noreferrer">Watch a video</a>
        </div>
      </div>
      <div className="card-body">
        <h4>Instructions: </h4>
        <p className="card-text">{food && food.strInstructions}</p>
      </div>
    </div>
  </div>
);

FoodsDetail.propTypes = {
  food: PropTypes.shape({
    idMeal: PropTypes.string.isRequired,
    strMeal: PropTypes.string.isRequired,
    strArea: PropTypes.string.isRequired,
    strCategory: PropTypes.string.isRequired,
    strInstructions: PropTypes.string.isRequired,
    strMealThumb: PropTypes.string.isRequired,
    strYoutube: PropTypes.string.isRequired,
    title: PropTypes.string,
    strIngredient1: PropTypes.string.isRequired,
    strIngredient2: PropTypes.string.isRequired,
    strIngredient3: PropTypes.string.isRequired,
    strIngredient4: PropTypes.string.isRequired,
  }).isRequired,
};

const mapStateToProps = (state, props) => ({
  food: state.foods.find((ff) => ff.idMeal === props.match.params.id),
});

const mapDispatchToProps = (dispatch) => ({
  updateFoods: (id, update) => dispatch(updateFoods(id, update)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FoodsDetail);
