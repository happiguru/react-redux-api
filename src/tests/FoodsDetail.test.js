import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import FoodsDetail from '../components/FoodDetail';
import TestStore from './store';

describe('Foods list', () => {
  test('should match with snapshot', () => {
    const jsx = (
      <Provider store={TestStore}>
        <Router>
          <FoodsDetail />
        </Router>
      </Provider>
    );
    const tree = (() => renderer.create(jsx).toJSON());
    expect(tree).toMatchSnapshot();
  });
});
