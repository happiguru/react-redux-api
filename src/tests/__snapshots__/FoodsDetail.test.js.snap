// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foods list should match with snapshot 1`] = `[Function]`;
